views/js/testRunner/testMetaData.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+/**
+ * Metadata to be sent to the server. Will be saved in result storage as a trace variable.
+ * Usage example:
+ * <pre>
+ * var testMetaData = new TestMetaData({
+ *   testServiceCallId : this.itemServiceApi.serviceCallId
+ * });
+ *
+ * testMetaData.setData({
+ *   'TEST' : {
+ *      'TEST_EXIT_CODE' : 'T'
+ *   },
+ *   'SECTION' : {
+ *      'SECTION_EXIT_CODE' : 704
+ *   }
+ * });
+ *
+ * testMetaData.addData({'ITEM' : {
+ *      'ITEM_START_TIME_CLIENT' : 1443596730143,
+ *      'ITEM_END_TIME_CLIENT' : 1443596731301
+ *    }
+ * });
+ * </pre>
+ *
+ * Data for each service call id will be stored in local storage to be able get data
+ * after reloading the page or resuming the test session.
+ *
+ * To clear all data related to current test_call_id used <i>clearData</i> method.
+ */
+define([
+    'lodash'
+], function (_) {
+    'use strict';
+
+    /**
+     *
+     * @param {Object} options
+     * @param {string} options.testServiceCallId - test call id.
+     * @constructor
+     */
+    function TestMetaData(options) {
```

Comment:
Instead of using the function scope to assign your values and then clone the data, you could do a factory : 

```js

var testMetaDataFactory = function testMetaDataFactory(options){
 
     var privateStuffs;

     var testMetaData = {
         getData : function(){},
         setData : function(){},
     };
     return testMetaData;
};
```